hera_cal/nucal.py

```diff
@@ -464,3 +495,147 @@ def __iter__(self):
+            If None, inferred from data_flags.freqs
+        data_is_redsol : bool, default=False
+            If True, data_file only contains unique visibilities for each baseline group.
+            In this case, gain_flags and tol are required and antpos is required if not derivable
+            from data_flags. In this case, the noise variance is inferred from autocorrelations from
+            all baselines in the represented unique baseline group.
+        gain_flags : dictionary, default=None
+            Used to exclude ants from the noise variance calculation from the autocorrelations
+            Ignored if data_is_redsol is False.
+        tol : float, 
+            Distance for baseline match tolerance in units of baseline vectors (e.g. meters).
+            Ignored if data_is_redsol is False.
+        antpos : dictionary
+            Maps antenna number to ENU position in meters for antennas in the data.
+            Ignored if data_is_redsol is False. If left as None, can be inferred from data_flags.data_antpos.
+        min_u_cut : float
+            Minimum u-magnitude value to include in calbration. All u-modes with magnitudes less than
+            min_u_cut will have their weights set to 0.
+        max_u_cut : float
+            Maximum u-magnitude value to include in calbration. All u-modes with magnitudes greater than
+            max_u_cut will have their weights set to 0.
+        min_freq_cut : float
+            Minimum frequency value to include in calibration in units of Hz. All frequency channels less than
+            this value will be set to 0.
+        max_freq_cut : float
+            Maximum frequency value to include in calibration in units of Hz. All frequency channels greater than
+            this value will be set to 0.
+        spw_range_flags : list of tuples
+            List of tuples containing the start and stop frequency of each spectral window to flag in units of Hz.
+                 
+    Returns:
+    -------
+        wgts: Datacontainer
+            Maps data_flags baseline to weights
+    """
+    # Build model flags from 
+    model_flags = {}
+    for group in radial_reds:
+        for key in group:
+            # Get u-magnitudes of all samples for this baseline
+            umag = radial_reds.baseline_lengths[key] * freqs / SPEED_OF_LIGHT
+            wgts = np.zeros_like(data_flags[key], dtype=bool)
+
+            # Apply u-magnitude and frequency cuts
+            if min_u_cut is not None:
+                wgts[:, umag < min_u_cut] = True
+            if max_u_cut is not None:
+                wgts[:, umag > max_u_cut] = True
+            if min_freq_cut is not None:
+                wgts[:, freqs < min_freq_cut] = True
+            if max_freq_cut is not None:
+                wgts[:, freqs > max_freq_cut] = True
+            if spw_range_flags is not None:
+                for spw in spw_range_flags:
+                    wgts[:, (freqs > spw[0]) & (freqs < spw[1])] = True
```

Comment:
Shouldn't these `True`s be 0s?

Reply:
Changed the name of the variable to `flags` so this shouldn't be a problem now